### PR TITLE
EVA-2150 - Changes to support the ticket

### PR DESCRIPTION
**Pull request by sundarvenkata-EBI**
- Created separate top-level package **variation** for domain-specific utilities
- assembly_utils.py - Add function to get assembly report URL
- contig_utils.py - Add function to resolve contig accession to chromosome names.

### CODE REVIEW

**Comment by sundarvenkata-EBI** [comment]
@tcezard  @andresfsilva Github automatically closed the original PR due to a rebase so I had to open this one :( Anyway, I have addressed @tcezard's  review comments from the original one.

File: ebi_eva_common_pyutils/variation/contig_utils.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+import re
+import requests
+from retry import retry
+
+
+# TODO: Might be a good idea to re-visit this after a production implementation
+#  of the contig-alias resolution project is available
+@retry(tries=10, delay=5, backoff=1.2, jitter=(1, 3))
+def resolve_contig_accession_to_chromosome_name(contig_accession, line_limit=100):
```
Comment (tcezard) [inline ebi_eva_common_pyutils/variation/contig_utils.py:23]: Are we sure this function will provide information that is not already in the assembly report.
This code looks very complicated for simply finding an alias and I'm not sure we really need it.

Comment (sundarvenkata-EBI) [reply ebi_eva_common_pyutils/variation/contig_utils.py:23]: Yes. This is the python-version of the logic that is currently being used in the website and has been reviewed previously. In fact, we explicitly reach for this in EVA-2150 detection script only when [we are out of luck with finding the chromosome name in the assembly report](https://github.com/EBIvariation/eva-tasks/pull/12/files#diff-8fee2a738e1f8916fe0967dc0af81c14R93).

File: ebi_eva_common_pyutils/variation/assembly_utils.py [hunk 2]
```diff
@@ -53,3 +55,35 @@ def resolve_assembly_name_to_GCA_accession(assembly_name):
...
+    ftp.cwd(genome_subfolders[0])
+    genome_files = []
+    ftp.retrlines('NLST', lambda line: genome_files.append(line))
+    ftp.quit()
+
+    assembly_reports = [genome_file for genome_file in genome_files if 'assembly_report' in genome_file]
+    if len(assembly_reports) != 1:
+        raise Exception('more than one file has "assembly_report" in its name: ' + str(assembly_reports))
+
+    return 'ftp://' + 'ftp.ncbi.nlm.nih.gov' + '/' + genome_folder + genome_subfolders[0] + '/' + assembly_reports[0]
```
[the remark this answers is not in the document]
Comment (sundarvenkata-EBI) [reply ebi_eva_common_pyutils/variation/assembly_utils.py:89]: Sure, we can coordinate the rebase and merge.

**Review by tcezard — APPROVED** [review]

**Review by andresfsilva — APPROVED** [review]

**Pull request merged**